src/components/Articles/Articles.js

```diff
@@ -1,51 +1,98 @@
-import React, {useEffect, useState} from 'react'
+import React, {useEffect, useState, useRef} from 'react'
 import {Link} from 'react-router-dom';
-import profileImg from '../../assets/profilepic.png'
-import articleImg from '../../assets/articles.jpg'
 import './Articles.css'
 import * as Requests from '../../Requests'
+import SearchBar from './SearchBar'
 
-const Articles = () => {
+
+const Articles = (props) => {
+
+  
     const [articles, setArticles] = useState([]);
+    const [searchTerm, setSearchTerm] = useState('');
+    const [searchResults, setSearchResults] = useState([]);
 
     useEffect(()=>{
-        Requests.getAllArticles().then(data=>{
-            if(data.statusCode !== 404) {
-                setArticles(data)
-            }
-        }).catch(() => null);
-    },[]);
-    return (
-        <div className = "full-page">
-            <div className = "articles-container">
+      Requests.getAllArticles().then(data=>{
+          if(data.statusCode !== 404) {
+              setArticles(data)
+          }
+      }).catch(() => null);
+  },[]);
+    const searchHandler = (searchTerm) => {
+      setSearchTerm(searchTerm);
+      if (searchTerm !== ""){
+        const newArticleList = articles.filter((article) => {
+          return Object.values(article).join(" ").toLowerCase().includes(searchTerm.toLowerCase()) ;
           
+        })
+        setSearchResults(newArticleList);
```

Comment:
Instead of using setSearchResults, try setArticles function. That way you don't need to have duplicate html below.